### PR TITLE
Group private methods in the app file

### PR DESCRIPTION
**Pull request by jorgegoco**
### In this PR, we : 

- Made sure YAGNI, KISS, and DRY rules and Ruby style guides were respected.
- Followed SOLID principles
- Paid special attention to the "Single-responsibility Principle".
- Made sure that the role of main.rb is only to start our code and ask the user for the initial choice of options (or to quit).
- Extract some pieces of logic into a separate class(es).

### CODE REVIEW

File: menu.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+require_relative './app'
```
Comment (arslanbisharat) [inline menu.rb]: - [ ] The role of `main.rb` is only to start your code and ask the user for the initial choice of options (or to quit). You should move the content of `menu.rb` to `main.rb`

File: app.rb [hunk 2]
```diff
@@ -121,5 +88,38 @@ def show_rental_by_id
...
+      @people.push(Student.new(nil, age, name, parent_permission: true))
+      puts 'Person created successfully'
+    when 'N'
+      @people.push(Student.new(nil, age, name, parent_permission: false))
+      puts 'Person created successfully'
+    else
+      puts 'Invalid Input'
+    end
+  end
+
```
Comment (arslanbisharat) [inline app.rb]: - [ ] Kindly make sure you follow the guidelines given at `https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid`. Currently the code is not using best practices.

File: app.rb [hunk 3]
```diff
@@ -121,5 +88,38 @@ def show_rental_by_id
...
+    puts 'Name:'
+    name = gets.chomp.to_s
+    puts 'Specialization:'
+    specialization = gets.chomp.to_s
+    @people.push(Teacher.new(specialization, age, name))
+    puts 'Person created successfully'
+  end
+
+  def index_out_of_bounds?(idx, array)
+    idx >= array.length || idx.negative?
```
Comment (arslanbisharat) [inline app.rb]: - [ ] Kindly make sure you follow the guidelines given at `https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid`. Currently the code is not using best practices.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @contributor_name, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️
- There is no linter error is present. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @jorgegoco and @maaloum,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**